### PR TITLE
Enable the update of archive policy back window

### PR DESCRIPTION
**Pull request by rafaelweingartner**

### CODE REVIEW

**Comment by rafaelweingartner** [comment]
Errors are not related to the PR

**Comment by rafaelweingartner** [comment]
This PR depends on the merge of https://github.com/gnocchixyz/gnocchi/pull/1279

**Comment by tobias-urdin** [comment]
@mergifyio rebase

File: gnocchi/indexer/sqlalchemy.py [hunk 1]
```diff
@@ -675,6 +675,8 @@ def update_archive_policy(self, name, ap_items):
                         % utils.timespan_total_seconds(c.granularity))
             # NOTE(gordc): ORM doesn't update JSON column unless new
             ap.definition = ap_items
+            if kwargs.get("back_window") is not None:
```
Comment (tobias-urdin) [inline gnocchi/indexer/sqlalchemy.py:678]: kwargs.get("back_window", None)

File: gnocchi/rest/api.py [hunk 2]
```diff
@@ -275,7 +277,8 @@ def patch(self):
 
         try:
             return pecan.request.indexer.update_archive_policy(
-                self.archive_policy, ap_items)
+                self.archive_policy, ap_items,
+                back_window=body.get('back_window'))
```
Comment (tobias-urdin) [inline gnocchi/rest/api.py:281]: Same here maybe?
body.get('back_window', None)

**Review by tobias-urdin — CHANGES_REQUESTED** [review]
I would be OK merging this in master but it need to update some testing and documentation as well. Also added a two comments that would be good to adress to prevent possible KeyError's.

File: gnocchi/rest/api.py (hunk 2 above)
Comment (rafaelweingartner) [reply gnocchi/rest/api.py:281]: Same here

File: gnocchi/indexer/sqlalchemy.py (hunk 1 above)
Comment (tobias-urdin) [reply gnocchi/indexer/sqlalchemy.py:678]: Ok, good 👍

**Comment by rafaelweingartner** [comment]
@tobias-urdin thanks for the review. I added tests and documentation as you requested.

**Review by tobias-urdin — APPROVED** [review]

**Pull request merged**